frontend/src/features/manageVehicles/apiManager/hooks.ts

```diff
@@ -11,12 +11,20 @@ import {
   updatePowerUnit,
   updateTrailer,
 } from "./vehiclesAPI";
+import { getCompanyIdFromSession } from "../../../common/apiManager/httpRequestHandler";
+import { Nullable } from "../../../common/types/common";
+import { getDefaultRequiredVal } from "../../../common/helpers/util";
 
 /**
  * Fetches all vehicles.
  * @returns An array of vehicles (both Powerunits and Trailers)
  */
-export const useVehiclesQuery = (companyId: string) => {
+export const useVehiclesQuery = (companyIdParam?: Nullable<string>) => {
+  const companyId = getDefaultRequiredVal(
+    "",
+    getCompanyIdFromSession(),
```

Comment:
um ok I guess, as long as we can guarantee that the companyId in the sessionStorage is always correct and up-to-date (since a non-empty stale value could cause problems). Or is it possible to call getCompanyIdFromSession() from elsewhere and pass the value in as companyIdParam instead?

Reply:
Got it, I guess we could try this approach